### PR TITLE
Readme update - remove optional/invalid builder methods from the `Usage` section

### PR DESCRIPTION
**Pull request by JakubJankowski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,37 +96,34 @@ The Asset Publisher SDK can efficiently acquire the location data on a device an
...
 import AblyAssetTrackingPublisher
 
-// Initialise a Publisher
+// Initialise a Publisher using the mandatory builder methods
 
 publisher = try? PublisherFactory.publishers() // get a Publisher Builder
     .connection(ConnectionConfiguration(apiKey: ABLY_API_KEY, clientId: CLIENT_ID)) // provide Ably configuration with credentials
-    .log(LogConfiguration()) // provide logging configuration
-    .transportationMode(TransportationMode()) // provide mode of transportation for better location enhancements
-    .delegate(self) // provide delegate to handle location updates locally if needed
```
Comment (KacperKluka) [inline README.md:105]: Even though this is optional I'd leave the `delegate()` in the example. Unless that's super obvious for iOS developers, then we can remove it (as an Android dev it wouldn't be obvious to me :sweat_smile: )

File: README.md [hunk 2]
```diff
@@ -96,37 +96,34 @@ The Asset Publisher SDK can efficiently acquire the location data on a device an
...
 
-// Initialise a Subscriber
+// Initialise a Subscriber using the mandatory builder methods
 
 subscriber = try? SubscriberFactory.subscribers() // get a Subscriber Builder
     .connection(ConnectionConfiguration(apiKey: ABLY_API_KEY, clientId: CLIENT_ID)) // provide Ably configuration with credentials
     .trackingId(trackingId) // provide a Tracking ID for the asset to be tracked
-    .routingProfile(.cycling) // provide a routing profile for better location enhancements
-    .log(LogConfiguration()) // provide logging configuration
-    .delegate(self) // provide a delegate to handle received location updates
```
Comment (KacperKluka) [inline README.md:127]: Like above, unless it's obvious to iOS devs and can be omitted :wink:

**Review by KacperKluka — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (JakubJankowski) [reply README.md:105]: Hmm yeah, I had an internal fight about this as well. I've decided to stick to the "mandatory methods only" and I left a comment below about the protocols with optional methods, but it's probably not enough. I'll re-add those delegates then 👍

Comment (JakubJankowski) [reply README.md:105]: [d2166b9](https://github.com/ably/ably-asset-tracking-swift/pull/332/commits/d2166b9b343c509726c2488ad2fd23ff62791938)

File: README.md (hunk 2 above)
Comment (JakubJankowski) [reply README.md:127]: [d2166b9](https://github.com/ably/ably-asset-tracking-swift/pull/332/commits/d2166b9b343c509726c2488ad2fd23ff62791938)

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Review by arturjaworski — APPROVED** [review]

**Pull request merged**